flink-connector-bigquery-common/src/main/java/com/google/cloud/flink/bigquery/common/utils/AvroToBigQuerySchemaTransform.java

```diff
@@ -0,0 +1,456 @@
+
+        return Field.newBuilder(name, dataType).setMode(Field.Mode.REQUIRED).build();
+    }
+
+    /**
+     * Helper method to handle Avro decimal logical types by determining the appropriate BigQuery
+     * data type based on the precision of the decimal.
+     *
+     * @param avroSchema the Avro schema of the decimal field
+     * @param name the name of the field
+     */
+    private static StandardSQLTypeName handleDecimalLogicalType(
+            org.apache.avro.Schema avroSchema, String name) {
+        long precision = validatePrecisionAndScale(avroSchema, name);
+        long scale = ((LogicalTypes.Decimal) avroSchema.getLogicalType()).getScale();
+
+        if (precision > 0 && precision <= (scale + 29) && scale <= 9) {
+            return StandardSQLTypeName.NUMERIC;
+        } else if (precision > 0 && precision <= (scale + 38) && scale <= 38) {
+            return StandardSQLTypeName.BIGNUMERIC;
+        }
+        LOG.warn(
+                "The precision {} and scale {} of decimal field {} is not supported by BigQuery. Converting the field to type STRING.",
+                precision,
+                scale,
+                name);
+        return StandardSQLTypeName.STRING;
+    }
+
+    /**
+     * Validates the precision and scale of an Avro decimal logical type. Ensures precision and
+     * scale are non-negative and that scale does not exceed precision.
+     *
+     * @param schema the Avro schema of the decimal field
+     * @param name the name of the field
+     * @return the precision of the decimal
+     * @throws IllegalArgumentException if precision or scale are invalid
+     */
+    private static long validatePrecisionAndScale(org.apache.avro.Schema schema, String name) {
+        LogicalTypes.Decimal decimalLogicalSchema =
+                ((LogicalTypes.Decimal) schema.getLogicalType());
+        Preconditions.checkNotNull(
+                decimalLogicalSchema.getPrecision(),
+                "The avro schema of type \"decimal\" must have the precision attribute set.");
+        long precision = decimalLogicalSchema.getPrecision();
+        long scale = decimalLogicalSchema.getScale();
+        if (precision <= 0) {
+            throw new IllegalArgumentException(
+                    "Precision of decimal field"
+                            + name
+                            + "must be non-negative. Saw: "
+                            + precision);
+        }
+        if (scale < 0) {
+            throw new IllegalArgumentException(
+                    "Scale of decimal field" + name + "must be non-negative. Saw: " + scale);
+        }
+        if (precision < scale) {
+            throw new IllegalArgumentException(
+                    "Scale of the field "
+                            + name
+                            + "cannot exceed precision. Saw scale: "
+                            + scale
+                            + ", precision: "
+                            + precision);
+        }
```

Comment:
Please use string format instead of + concat

Reply:
Done